st2common/st2common/models/db/__init__.py

```diff
@@ -113,7 +115,9 @@ def get_all(self, *args, **kwargs):
         return self.query(*args, **kwargs)
 
     def count(self, *args, **kwargs):
-        return self.model.objects(**kwargs).count()
+        result = self.model.objects(**kwargs).count()
+        log_query_and_profile_data_for_queryset(queryset=result)
```

Comment:
My original plan was to monkey patch mongoengine `QuerySet.__call__` method, but it turns out this doesn't work too well with the chained methods approach (e.g. `foo.query().order_by().limit()...`).

In any case, luckily we have an additional abstraction layer for talking to the database so we can just add this instrumentation here. Some people might also argue that this approach is also better since it's more explicit.